### PR TITLE
fix: Fix condition for choosing runner

### PR DESCRIPTION
**Pull request by toninis**
#### Summary

We will use default runners for contributor Pull Requests . If this does not work we will revert the changes.
**Plus:** Automate the image choice 😄 

#### Release Note

```release-note

```

### CODE REVIEW

File: .github/workflows/server-test-template.yml [hunk 1]
```diff
@@ -56,6 +56,8 @@ jobs:
           docker compose --ansi never exec -T minio sh -c 'mkdir -p /data/mattermost-test';
           docker compose --ansi never ps
       - name: Run Tests
+        env:
+          BUILD_IMAGE: mattermost/mattermost-build-server:${{ steps.go.outputs.GO_VERSION }}
```
Comment (isacikgoz) [inline .github/workflows/server-test-template.yml:60]: Hmm, what would happen if there is no build-server corresponding to the `GO_VERSION`?

Comment (toninis) [reply .github/workflows/server-test-template.yml:60]: It will fail . But I updated the upgrade golang version to fix the images prior changing this . 
We also use the same in our packaging pipeline [here](https://github.com/mattermost/delivery-platform/blob/master/.github/workflows/mattermost-platform-package-dev.yml#L43)

**Review by isacikgoz — APPROVED** [review]

File: .github/workflows/server-test-template.yml (hunk 1 above)
Comment (agnivade) [inline .github/workflows/server-test-template.yml:60]: Why is this moved down to only this step?

**Review by agnivade — APPROVED** [review]
Left a small comment.

File: .github/workflows/server-test-template.yml (hunk 1 above)
Comment (toninis) [reply .github/workflows/server-test-template.yml:60]: This is the only step it is used so we do not need it globally declared in the job. 
Also I want to utilize the output variable and I cannot in the job env context  since the output is generated throughout the job
